wasmplugin/plugin.go

```diff
@@ -119,18 +140,32 @@ func (ctx *corazaPlugin) OnPluginStart(pluginConfigurationSize int) types.OnPlug
 			return types.OnPluginStartStatusFailed
 		}
 
-		err = perAuthorityWAFs.put(name, waf)
-		if err != nil {
-			proxywasm.LogCriticalf("Failed to register authority WAF: %v", err)
-			return types.OnPluginStartStatusFailed
+		if setDefaultWAF {
+			if err := perAuthorityWAFs.setDefaultWAF(waf); err != nil {
+				proxywasm.LogCriticalf("Failed to set default WAF: %v", err)
```

Comment:
Reason for above panic suggestion is propagating errors for this sort of case makes the code very confusing

Reply:
Done.